openshift/kustomize/tekton/base/tasks/deploy-all.yaml

```diff
@@ -364,10 +324,18 @@ spec:
         done
 
         # Start all services if API was deployed.
-        if [ "${BUILD_API}" = true ] || [ "$(params.DEPLOY)" = "api" ] || [ "$(params.DEPLOY)" = "all" ]; then
+        if [ "${BUILD_API}" = true ] || [ "$(params.DEPLOY)" = "api" ] || [ "$(params.DEPLOY)" = "all" ] || [ "$(params.DEPLOY)" = "${component[id]}" ]; then
           for component in ${!COMPONENT@}; do
+            # Dynamic variable to control what will be built.
+            TVAR="BUILD_${component[id]}"
```

Comment:
Fixed bug where it wasn't using the correct component configuration.  This resulted in failures if an environment did not have the specified component.  For example the DEV environment doesn't have an index-service-cloud component because there is no Elasticsearch Cloud deployment to link to.